snap/snapcraft.yaml

```diff
@@ -0,0 +1,283 @@
+  /usr/share/openmpi:
+    symlink: $SNAP/usr/share/openmpi
+  /etc/openmpi:
+    bind: $SNAP/etc/openmpi
+  /usr/lib/$SNAPCRAFT_ARCH_TRIPLET/openmpi:
+    bind: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/openmpi
+  /usr/bin/orted:
+    symlink: $SNAP/usr/bin/orted
+  /etc/matplotlibrc:
+    bind-file: $SNAP/etc/matplotlibrc
+  /usr/share/matplotlib:
+    symlink: $SNAP/usr/share/matplotlib
+  /usr/share/qt5:
+    symlink: $SNAP/kf5/usr/share/qt5
+
+environment:
+  LD_LIBRARY_PATH: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/blas:$SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/lapack # numpy
+  LD_PRELOAD: $SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/libstubchown.so
+  FREECAD_USER_HOME: $SNAP_USER_COMMON
+  GIT_EXEC_PATH: $SNAP/usr/lib/git-core
+  GIT_TEMPLATE_DIR: $SNAP/usr/share/git-core/templates
+  GIT_CONFIG_NOSYSTEM: 1
+  ELMER_HOME: $SNAP/usr
+  QTWEBENGINE_DISABLE_SANDBOX: 1
+  PIP_TARGET: $SNAP_USER_COMMON/snap-user-packages
+  PYTHONPATH: $SNAP_USER_COMMON/snap-user-packages:$SNAP/lib/python3.8/site-packages:$SNAP/usr/lib/python3/dist-packages
+
+apps:
+  freecad-realthunder:
+    command: freecad-launcher.sh
+    extensions: [kde-neon]
+    common-id: org.freecadweb.FreeCAD.desktop
+    desktop: usr/share/applications/org.freecadweb.FreeCAD.desktop
+    plugs: &plugs
+      - home
+      - opengl
+      - removable-media
+      - gsettings
+      - network      
+      - browser-support
+  cmd:
+    command: usr/bin/FreeCADCmd
+    extensions: [kde-neon]
+    plugs: *plugs
+  pip:
+    command: bin/pip
+    plugs: *plugs
+
+package-repositories:
+  - type: apt
+    ppa: elmer-csc-ubuntu/elmer-csc-ppa
+  - type: apt
+    ppa: openscad/releases
+  - type: apt
+    ppa: freecad-maintainers/occt-releases
+
+parts:
+  stub-chown:
+    plugin: meson
+    source: snap/local/stub-chown
+    meson-parameters:
+      - --prefix=/usr
+
+  launcher:
+    plugin: dump
+    source: snap/local/launcher
+    stage-packages:
+      - xterm # for py_slvs install prompt
+
+  coin3d:
+    plugin: cmake
+    source: https://github.com/realthunder/coin.git
+    source-tag: Coin-20210915 
+    cmake-parameters:
+      - -DCMAKE_INSTALL_PREFIX=/usr
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DCOIN_BUILD_TESTS=OFF
+      - -DCMAKE_CXX_STANDARD=98
+
+  pivy:
+    after: [coin3d]
+    plugin: cmake
+    source: https://github.com/coin3d/pivy.git
+    source-tag: "0.6.6"
+    override-build: |
+      # out of source builds fail for some reason:
+      # https://github.com/coin3d/pivy/issues/72
+      # let's do an in-source build  instead
+      cd $SNAPCRAFT_PART_SRC
+      cmake \
+        -DCMAKE_FIND_ROOT_PATH=/snap/kde-frameworks-5-qt-5-15-3-core20-sdk/current \
+        -DCMAKE_PREFIX_PATH="$SNAPCRAFT_STAGE" \
+        -DCMAKE_INSTALL_PREFIX=/usr \
+        -DCMAKE_BUILD_TYPE=Release \
+        .
+      cmake --build . -- -j${SNAPCRAFT_PARALLEL_BUILD_COUNT}
+      DESTDIR="$SNAPCRAFT_PART_INSTALL" cmake --build  . --target install
```

Comment:
I decided to also compile pivy from source because the ubuntu package pulls in libcoin* from the repo, and I'd have to explicitly exclude it again. Let's keep it simple.